pymc3/step_methods/arraystep.py

```diff
@@ -156,6 +156,30 @@ def step(self, point):
             return bij.rmap(apoint)
 
 
+class GradientSharedStep(BlockedStep):
+    def __init__(self, vars, model=None, blocked=True,
+                 gpu_ctx=None, dtype=None, **theano_kwargs):
+        model = modelcontext(model)
+        self.vars = vars
+        self.blocked = blocked
+
+        self._logp_dlogp_func = model.logp_dlogp_function(
+            vars, dtype=dtype, **theano_kwargs)
+
+    def step(self, point):
+        self._logp_dlogp_func.set_extra_values(point)
+        array = self._logp_dlogp_func.dict_to_array(point)
+
+        if self.generates_stats:
+            apoint, stats = self.astep(array)
```

Comment:
When I wrote this I though I'd rather not change the interface for step methods, so that we don't break custom step methods. Now I guess we should probably treat that as private part of the api, so yes, I agree.